sdk/rust/oak/src/lib.rs

```diff
@@ -154,23 +154,25 @@ pub fn channel_read(half: ReadHandle, buf: &mut Vec<u8>, handles: &mut Vec<Handl
             ) as i32
         });
         match status {
-            Some(OakStatus::OK) => {
-                unsafe {
-                    buf.set_len(actual_size as usize);
-                    // Handles are written in little-endian order, which matches Wasm spec.
-                    handles.set_len(actual_handle_count as usize);
-                };
-                return OakStatus::OK;
-            }
-            Some(OakStatus::ERR_BUFFER_TOO_SMALL) => {
-                if *resized {
-                    return OakStatus::ERR_BUFFER_TOO_SMALL;
+            Some(s) => match s {
+                OakStatus::OK | OakStatus::ERR_CHANNEL_EMPTY => {
+                    unsafe {
+                        buf.set_len(actual_size as usize);
+                        // Handles are written in little-endian order, which matches Wasm spec.
+                        handles.set_len(actual_handle_count as usize);
+                    };
+                    return s;
                 }
-                // Can escape the match if buffer is too small and !resized.
-            }
-            Some(s) => {
-                return s;
-            }
+                OakStatus::ERR_BUFFER_TOO_SMALL => {
+                    if *resized {
+                        return s;
+                    }
+                    // Can escape the match if buffer is too small and !resized.
```

Comment:
(preexisting) Perhaps this should be split into two cases:

~~~rust
OakStatus::ERR_BUFFER_TOO_SMALL if !(*resized) => {
  // do resize here instead of outside the case below
}
OakStatus::ERR_BUFFER_TOO_SMALL => {
  panic!("Failed to resize buffer during channel_read!");
}
~~~

As the resize should always succeed unless there is a bug in our code?

Reply:
That's neat, I forgot about match guards, done. 

If `BUFFER_TOO_SMALL` comes up again, I went for passing it out to the caller, mostly because it makes the match expression even simpler/clearer.